examples/language-modeling/main.py

```diff
@@ -162,7 +166,51 @@
     torch_device = torch.device(device_str)
 
     is_glm = bool(re.search("chatglm", model_name.lower()))
-    if is_glm:
+    if args.layer_wise:
+        from auto_round.layer_wise.utils import load_model_with_hooks
+        if is_glm:
+            model = load_model_with_hooks(
+                model_name,
+                AutoModel,
+                device=device_str,
+                clean_weight=True,
```

Comment:
the common part is a little long, consilidate it with "model_type = AutoModel, load_model_xxx(model_type)"